src/esi/ExpatParser.cc

```diff
@@ -12,8 +12,30 @@
 
 #if USE_SQUID_ESI && HAVE_LIBEXPAT
 
+#include "base/RunnersRegistry.h"
 #include "esi/ExpatParser.h"
 
+#include <memory>
+
+namespace Esi
+{
+
+class ExpatRr : public RegisteredRunner
+{
+public:
+    void finalizeConfig()
+    {
+        prExpat.reset(new ESIParser::Register("expat", &ESIExpatParser::NewParser));
+    }
+
+private:
+    std::unique_ptr<ESIParser::Register> prExpat;
```

Comment:
We should not use unique_ptr for objects _shared_ with others. This Register object (i.e. the ESI parser registration object) is shared with the ESIParser registry (i.e. the std::list container accessible via ESIParser::GetRegistry()).

If you would like to clean this up, I can detail how to do that. However, this cleanup can be legitimately viewed as being outside this PR scope -- the official code has essentially the same flaw (unrelated to the module initialization/cleanup sequence itself) -- so I cannot insist on it.

Reply:
Left out of scope.